processor/resourcedetectionprocessor/README.md

```diff
@@ -406,6 +407,14 @@ processors:
     override: false
 ```
 
+#### Cluster Name
+
+Azure AKS cluster name is derived from the Azure Instance Metadata Service's (IMDS) infrastructure resource group field. This field contains the resource group and name of the cluster, separated by underscores. e.g: `MC_<resource group>_<cluster name>_<location>`.
+
+The cluster name is accurately detected if underscores are not present in the resource group name or the cluster name.
```

Comment:
Imo we should clarify that we can only get the correct cluster name if custom resource group name is not used. I still think some users might prefer setting the clusterName themself then defaulting to custom resource group name. So explicitly making a note of what is returned in different scenarios would provide better user experience.

Reply:
Agreed. Apologies for the late response.